File: beginners_selection/abc083b.py
# https://atcoder.jp/contests/abs/tasks/abc083_b

n, a, b = map(int, input().split())

array = []
for i in range(1, n+1):
    str_i = str(i)

    digit_sum = 0
    if len(str_i) >= 2:
        for s in str_i:
            digit_sum += int(s)
    else:
        digit_sum = i
    
    if a <= digit_sum and digit_sum <= b:
        array.append(i)

print(sum(array))

File: cpp/beginners_selection/abc083b.cpp
// https://atcoder.jp/contests/abs/tasks/abc083_b

#include<iostream>
using namespace std;

int ctoi(const char c){
  if('0' <= c && c <= '9') return (c-'0');
  return -1;
}

int main() {
    int n, a, b;
    cin >> n >> a >> b;

    int sum = 0;
    for (int i = 1; i <= n; i++) {
        int num = 0;
        string s = to_string(i);
        for(char& c: s) {
            num += ctoi(c);
        }
        if (a <= num && num <= b) {
            sum += i;
        }
    }

    cout << sum << endl;
    return 0;
}
